### ISSUE TITLE
Running migrations to 19.6.0 fails - Cannot read properties of undefined (reading 'data')

### ISSUE THREAD

**Issue by schreibse** [nrwl/nx#27470]
### Current Behavior

After migration to latest nx version `nx migrate latest` (19.6.0), the command to run the migrations `nx migrate --run-migrations` fails, please see failure logs section. 

### Expected Behavior

`nx migrate --run-migrations ` finishes without error

### Steps to Reproduce

1. In a repo with nx 19.5.7 run `nx migrate latest`
2. Once packge.json is updated and migrations file is added, run `nx migrate --run-migrations`

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 20.14.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.7.0

nx (global)        : 19.2.3
nx                 : 19.6.0
@nx/js             : 19.6.0
@nx/jest           : 19.6.0
@nx/linter         : 19.6.0
@nx/eslint         : 19.6.0
@nx/workspace      : 19.6.0
@nx/angular        : 19.6.0
@nx/devkit         : 19.6.0
@nx/eslint-plugin  : 19.6.0
@nx/playwright     : 19.6.0
@nrwl/tao          : 19.6.0
@nx/vite           : 19.6.0
@nx/web            : 19.6.0
@nx/webpack        : 19.6.0
typescript         : 5.5.3
---------------------------------------
Registered Plugins:
@nx/playwright/plugin
@nx/eslint/plugin
---------------------------------------
Community plugins:
@jsverse/transloco : 7.4.3
@ngrx/operators    : 18.0.1
@ngrx/signals      : 18.0.0
eslint-plugin-ngrx : 2.1.4
```

### Failure Logs

```shell
NX   Failed to run 19-6-0-use-serve-static-preview-for-command from @nx/playwright. This workspace is NOT up to date!

 NX   Cannot read properties of undefined (reading 'data')

TypeError: Cannot read properties of undefined (reading 'data')
    at addE2eCiTargetDefaults (C:\Dev\XXX\XXX\XXX\node_modules\@nx\playwright\src\migrations\update-19-6-0\add-e2e-ci-target-defaults.js:68:79)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async default_1 (C:\Dev\XXX\XXX\XXX\node_modules\@nx\playwright\src\migrations\update-19-6-0\use-serve-static-preview-for-command.js:123:5)
    at async runNxMigration (C:\Dev\XXX\XXX\XXX\node_modules\nx\src\command-line\migrate\migrate.js:1036:5)
    at async executeMigrations (C:\Dev\XXX\XXX\XXX\node_modules\nx\src\command-line\migrate\migrate.js:921:33)
    at async runMigrations (C:\Dev\XXX\XXX\XXX\node_modules\nx\src\command-line\migrate\migrate.js:1009:37)
    at async C:\Dev\XXX\XXX\XXX\node_modules\nx\src\command-line\migrate\migrate.js:1053:13
    at async handleErrors (C:\Dev\XXX\XXX\XXX\node_modules\nx\src\utils\params.js:22:24)
    at async Object.handler (C:\Dev\XXX\XXX\XXX\node_modules\nx\src\command-line\migrate\command-object.js:26:43)
Command failed: npx nx _migrate --run-migrations --verbose

 NX   Command failed: npx nx migrate --run-migrations --verbose

Error: Command failed: npx nx migrate --run-migrations --verbose
    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at runNxSync (C:\Users\yyy\AppData\Local\Temp\tmp-31220-huQBV91j5M2B\node_modules\nx\src\utils\child-process.js:27:34)
    at runMigrations (C:\Users\yyy\AppData\Local\Temp\tmp-31220-huQBV91j5M2B\node_modules\nx\src\command-line\migrate\migrate.js:984:39)
    at C:\Users\yyy\AppData\Local\Temp\tmp-31220-huQBV91j5M2B\node_modules\nx\src\command-line\migrate\migrate.js:1053:19
    at async handleErrors (C:\Users\yyy\AppData\Local\Temp\tmp-31220-huQBV91j5M2B\node_modules\nx\src\utils\params.js:22:24)
Command failed: C:\Users\yyy\AppData\Local\Temp\tmp-31220-huQBV91j5M2B\node_modules\.bin\nx _migrate --run-migrations --verbose
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by hoeferbe** [comment]
My migration run also failed, but for @nx/cypress

```
Running migration @nx/cypress: update-19-6-0-update-ci-webserver-for-vite

 NX   Failed to run update-19-6-0-update-ci-webserver-for-vite from @nx/cypress. This workspace is NOT up to date!


 NX   matchingPluginRegistrations is not iterable


TypeError: matchingPluginRegistrations is not iterable
    at default_1 (/Users/hoeferbe/GIT/dua-monorepo/node_modules/@nx/cypress/src/migrations/update-19-6-0/update-ci-webserver-for-static-serve.js:14:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runNxMigration (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:1036:5)
    at async executeMigrations (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:921:33)
    at async runMigrations (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:1009:37)
    at async /Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:1053:13
    at async handleErrors (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/utils/params.js:22:24)
    at async Object.handler (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/command-object.js:26:43)
Command failed: npx nx _migrate --run-migrations --verbose

 NX   Command failed: npx nx migrate --run-migrations --verbose


Error: Command failed: npx nx migrate --run-migrations --verbose
    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at runNxSync (/private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/utils/child-process.js:27:34)
    at runMigrations (/private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/command-line/migrate/migrate.js:984:39)
    at /private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/command-line/migrate/migrate.js:1053:19
    at async handleErrors (/private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/utils/params.js:22:24)
````

**Comment by jeoffreybakker** [comment]
> My migration run also failed, but for @nx/cypress
> 
> ```
> Running migration @nx/cypress: update-19-6-0-update-ci-webserver-for-vite
> 
>  NX   Failed to run update-19-6-0-update-ci-webserver-for-vite from @nx/cypress. This workspace is NOT up to date!
> 
> 
>  NX   matchingPluginRegistrations is not iterable
> 
> 
> TypeError: matchingPluginRegistrations is not iterable
>     at default_1 (/Users/hoeferbe/GIT/dua-monorepo/node_modules/@nx/cypress/src/migrations/update-19-6-0/update-ci-webserver-for-static-serve.js:14:26)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async runNxMigration (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:1036:5)
>     at async executeMigrations (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:921:33)
>     at async runMigrations (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:1009:37)
>     at async /Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/migrate.js:1053:13
>     at async handleErrors (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/utils/params.js:22:24)
>     at async Object.handler (/Users/hoeferbe/GIT/dua-monorepo/node_modules/nx/src/command-line/migrate/command-object.js:26:43)
> Command failed: npx nx _migrate --run-migrations --verbose
> 
>  NX   Command failed: npx nx migrate --run-migrations --verbose
> 
> 
> Error: Command failed: npx nx migrate --run-migrations --verbose
>     at genericNodeError (node:internal/errors:984:15)
>     at wrappedFn (node:internal/errors:538:14)
>     at checkExecSyncError (node:child_process:890:11)
>     at execSync (node:child_process:962:15)
>     at runNxSync (/private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/utils/child-process.js:27:34)
>     at runMigrations (/private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/command-line/migrate/migrate.js:984:39)
>     at /private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/command-line/migrate/migrate.js:1053:19
>     at async handleErrors (/private/var/folders/mv/tmk3tvsn0fq_2bwj0vxl2bvh0000gn/T/tmp-59156-MuV52f2f0pub/node_modules/nx/src/utils/params.js:22:24)
> ```

Same here, it was easy to fix by making sure that on line 13 of the script `matchingPluginRegistrations` defaults to a empty array.

**Comment by BabakScript** [comment]
I still have the same issue when upgrading from 18.3.3 to 19.6.1:

```
Running migration @nx/playwright: 19-6-0-use-serve-static-preview-for-command
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

 NX   Failed to run 19-6-0-use-serve-static-preview-for-command from @nx/playwright. This workspace is NOT up to date!


 NX   Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.

Pass --verbose to see the stacktraces.

Command failed: pnpm exec nx _migrate --run-migrations

 NX   Command failed: pnpm exec nx migrate --run-migrations

Pass --verbose to see the stacktrace.
```

**Comment by the-ult** [comment]
```
Running migration @nx/playwright: 19-6-0-use-serve-static-preview-for-command
D:\work\expo\node_modules\@nx\playwright\src\migrations\update-19-6-0\use-serve-static-preview-for-command.js:54
            (0, devkit_1.joinPathFragments)(graph.nodes[project].data?.root, 'vite.config.ts'),
                                                                 ^

    at D:\work\expo\node_modules\@nx\playwright\src\migrations\update-19-6-0\use-serve-static-preview-for-command.js:54:66
    at visitNotIgnoredFiles (D:\work\expo\node_modules\@nx\devkit\src\generators\visit-not-ignored-files.js:31:13)
    at visitNotIgnoredFiles (D:\work\expo\node_modules\@nx\devkit\src\generators\visit-not-ignored-files.js:34:13)
    at visitNotIgnoredFiles (D:\work\expo\node_modules\@nx\devkit\src\generators\visit-not-ignored-files.js:34:13)
    at default_1 (D:\work\expo\node_modules\@nx\playwright\src\migrations\update-19-6-0\use-serve-static-preview-for-command.js:15:39) 
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runNxMigration (D:\work\expo\node_modules\nx\src\command-line\migrate\migrate.js:1036:5)
    at async executeMigrations (D:\work\expo\node_modules\nx\src\command-line\migrate\migrate.js:921:33)
    at async runMigrations (D:\work\expo\node_modules\nx\src\command-line\migrate\migrate.js:1009:37)
    at async D:\work\expo\node_modules\nx\src\command-line\migrate\migrate.js:1053:13
    
```

```
Node           : 20.16.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.1.0

nx                 : 19.6.1
@nx/js             : 19.6.1
@nx/jest           : 19.6.1
@nx/linter         : 19.6.1
@nx/eslint         : 19.6.1
@nx/workspace      : 19.6.1
@nx/angular        : 19.6.1
@nx/cypress        : 19.6.1
@nx/devkit         : 19.6.1
@nx/eslint-plugin  : 19.6.1
@nx/playwright     : 19.6.1
@nx/storybook      : 19.6.1
@nrwl/tao          : 19.6.1
@nx/vite           : 19.6.1
@nx/web            : 19.6.1
@nx/webpack        : 19.6.1
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/playwright/plugin
@nx/eslint/plugin
@nx/jest/plugin
---------------------------------------
Community plugins:
@compodoc/compodoc         : 1.1.25
@ng-bootstrap/ng-bootstrap : 17.0.0
@ngrx/operators            : 18.0.2
@ngrx/signals              : 18.0.2
@storybook/angular         : 8.2.9
@testing-library/angular   : 17.3.0
ng-mocks                   : 14.13.0
ngxtension                 : 4.0.0
```

**Comment by leosvelperez** [comment]
@BabakScript @the-ult thanks for the info!

The fix missed the migration you're reporting (it fixed others with a similar issue). I'm reopening this to track it and I'll send a fix for it.

**Comment by leosvelperez** [comment]
@BabakScript @the-ult could please share the contents of your `playwright.config.ts`?

**Comment by the-ult** [comment]
```ts
/* eslint-disable unicorn/prefer-module */

import { nxE2EPreset } from '@nx/playwright/preset';
import { defineConfig, devices } from '@playwright/test';

import { workspaceRoot } from '@nx/devkit';

// For CI, you may want to set BASE_URL to the deployed application.
const baseURL = process.env['BASE_URL'] ?? 'http://localhost:4200';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  ...nxE2EPreset(__filename, {
    testDir: './src',
  }),
  retries: 2,
  snapshotPathTemplate: '{testDir}/__snapshots__/{testFilePath}/{arg}{ext}',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  expect: {
    toHaveScreenshot: {
      threshold: 0.15,
      maxDiffPixelRatio: 0.025,
      maxDiffPixels: 0.2,
    },
    toMatchSnapshot: {
      threshold: 0.15,
      maxDiffPixelRatio: 0.025,
      // maxDiffPixels: 0.2,
    },
  },
  use: {
    baseURL,
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
    viewport: { width: 1920, height: 1080 },
  },
  /* Run your local dev server before starting the tests */
  webServer: {
    command: 'npx nx serve expo',
    url: 'http://localhost:4200',
    reuseExistingServer: !process.env['CI'],
    cwd: workspaceRoot,
  },
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'], viewport: { width: 1920, height: 1080 } },
    },
  ],
});

```

**Comment by leosvelperez** [comment]
Mmm, I can't reproduce the issue with that config as long as the `expo` project exists. Is that your only `playwright.config.ts` file? Or do you have other e2e projects? The main thing we're looking for is the `webServer.command` value.

That said, the migration needs to gracefully handle a wrong/invalid command pointing to a non-existing project or running a server that doesn't use Nx to run. I can fix that, but I'd like to understand first the exact issue you're having to rule out any other scenario we might be missing.

**Comment by BabakScript** [comment]
@leosvelperez Sure sure. I have two. This is the first one:

```
import { defineConfig } from '@playwright/test'
import { nxE2EPreset } from '@nx/playwright/preset'

import { workspaceRoot } from '@nx/devkit'

// For CI, you may want to set BASE_URL to the deployed application.
const baseURL = process.env['BASE_URL'] || 'http://localhost:4200'

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  ...nxE2EPreset(__filename, { testDir: './src' }),
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    baseURL,
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
  },
  /* Run your local dev server before starting the tests */
  webServer: {
    command: 'pnpm exec nx serve mysite.com',
    url: 'http://localhost:4200',
    reuseExistingServer: !process.env.CI,
    cwd: workspaceRoot,
  },
})

```

And:

```
import { defineConfig, devices } from '@playwright/test'
import { nxE2EPreset } from '@nx/playwright/preset'

import { workspaceRoot } from '@nx/devkit'

// For CI, you may want to set BASE_URL to the deployed application.
const baseURL = process.env['BASE_URL'] || 'http://localhost:4200'

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  ...nxE2EPreset(__filename, { testDir: './src' }),
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    baseURL,
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
  },
  /* Run your local dev server before starting the tests */
  webServer: {
    command: 'pnpm exec nx serve mysite2.com',
    url: 'http://localhost:4200',
    reuseExistingServer: !process.env.CI,
    cwd: workspaceRoot,
  },
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'] },
    },

    {
      name: 'firefox',
      use: { ...devices['Desktop Firefox'] },
    },

    {
      name: 'webkit',
      use: { ...devices['Desktop Safari'] },
    },

    // Uncomment for mobile browsers support
    /* {
      name: 'Mobile Chrome',
      use: { ...devices['Pixel 5'] },
    },
    {
      name: 'Mobile Safari',
      use: { ...devices['iPhone 12'] },
    }, */

    // Uncomment for branded browsers
    /* {
      name: 'Microsoft Edge',
      use: { ...devices['Desktop Edge'], channel: 'msedge' },
    },
    {
      name: 'Google Chrome',
      use: { ...devices['Desktop Chrome'], channel: 'chrome' },
    } */
  ],
})

```

**Comment by the-ult** [comment]
there is 1 other project.

Which has:

```
webServer: {
    command: 'npx nx serve hiep --port 4400',
    url: 'http://localhost:4400',
    reuseExistingServer: !process.env['CI'],
    cwd: workspaceRoot,
  },
  ``` 
  
  So the only difference seems to be the `--port`

**Comment by leosvelperez** [comment]
> So the only difference seems to be the `--port`

Yeah, I see the issue in the migration not considering extra flags after the project name.

> @leosvelperez Sure sure. I have two. This is the first one:

Thanks!

**Comment by leosvelperez** [comment]
@BabakScript I haven't been able to reproduce your issue. Would you be able to share your repo or create a repo where the issue can be reproduced?

**Comment by BabakScript** [comment]
@leosvelperez  My current repo is huge but I can prepare a repo to reproduce the issue, but give me time. I'll do it on the weekend.

**Comment by leosvelperez** [comment]
@BabakScript, we merged a fix that covers several things I found and "might" address the issue you're facing. Please try again once the fix is released (it should be included in v19.6.2).

**Comment by BabakScript** [comment]
@leosvelperez @FrozenPandaz  I checked with the latest version v.19.6.2 and I still have the same issue:

```
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

 NX   Failed to run 19-6-0-use-serve-static-preview-for-command from @nx/playwright. This workspace is NOT up to date!


 NX   Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.

Pass --verbose to see the stacktraces.
```

My report:
```
pnpm nx report
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

 NX   Report complete - copy this into the issue template

Node           : 20.14.0
OS             : linux-arm64
Native Target  : aarch64-linux
pnpm           : 9.7.1

nx                 : 19.6.2
@nx/js             : 19.6.2
@nx/jest           : 19.6.2
@nx/linter         : 19.6.2
@nx/eslint         : 19.6.2
@nx/workspace      : 19.6.2
@nx/devkit         : 19.6.2
@nx/esbuild        : 19.6.2
@nx/eslint-plugin  : 19.6.2
@nx/nest           : 19.6.2
@nx/node           : 19.6.2
@nx/playwright     : 19.6.2
@nx/react          : 19.6.2
@nrwl/tao          : 19.6.2
@nx/vite           : 19.6.2
@nx/web            : 19.6.2
@nx/webpack        : 19.6.2
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/webpack/plugin
@nx/eslint/plugin
@nx/vite/plugin
@nx/playwright/plugin
---------------------------------------
Community plugins:
@naxodev/nx-cloudflare : 2.2.0
```

**Comment by leosvelperez** [comment]
@BabakScript, your issue seems to be different than the one reported here, and we'll need a way to reproduce it. Please open a new issue and provide a reproduction. You can tag me and I'll take a look.

**Comment by danr-za** [comment]
My migration to 19.6.2 fails on Storybook v8 - it searches '@nx/react/plugins/storybook/package.json' due to 
```ts
getAbsolutePath('@nx/react/plugins/storybook')
```
